regression_tests/utils.go

```diff
@@ -0,0 +1,222 @@
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	walletclient "github.com/bitcoin-sv/spv-wallet-go-client"
+	"github.com/bitcoin-sv/spv-wallet-go-client/xpriv"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/bitcoin-sv/spv-wallet/models/filter"
+)
+
+const (
+	atSign                   = "@"
+	domainPrefix             = "https://"
+	domainSuffixSharedConfig = "/v1/shared-config"
+
+	ClientOneURLEnvVar         = "CLIENT_ONE_URL"
+	ClientTwoURLEnvVar         = "CLIENT_TWO_URL"
+	ClientOneLeaderXPrivEnvVar = "CLIENT_ONE_LEADER_XPRIV"
+	ClientTwoLeaderXPrivEnvVar = "CLIENT_TWO_LEADER_XPRIV"
+
+	timeoutDuration = 30 * time.Second
+)
+
+var (
+	explicitHTTPURLRegex = regexp.MustCompile(`^https?://`)
+	envVariableError     = errors.New("missing xpriv variables")
+)
+
+type regressionTestUser struct {
+	XPriv   string `json:"xpriv"`
+	XPub    string `json:"xpub"`
+	Paymail string `json:"paymail"`
+}
+
+type regressionTestConfig struct {
+	ClientOneURL         string
+	ClientTwoURL         string
+	ClientOneLeaderXPriv string
+	ClientTwoLeaderXPriv string
+}
+
+// getEnvVariables retrieves the environment variables needed for the regression tests.
+func getEnvVariables() (*regressionTestConfig, error) {
+	rtConfig := regressionTestConfig{
+		ClientOneURL:         os.Getenv(ClientOneURLEnvVar),
+		ClientTwoURL:         os.Getenv(ClientTwoURLEnvVar),
+		ClientOneLeaderXPriv: os.Getenv(ClientOneLeaderXPrivEnvVar),
+		ClientTwoLeaderXPriv: os.Getenv(ClientTwoLeaderXPrivEnvVar),
+	}
+
+	if rtConfig.ClientOneLeaderXPriv == "" || rtConfig.ClientTwoLeaderXPriv == "" {
+		return nil, envVariableError
+	}
+	if rtConfig.ClientOneURL == "" || rtConfig.ClientTwoURL == "" {
+		rtConfig.ClientOneURL = "http://localhost:3003"
+		rtConfig.ClientTwoURL = "http://localhost:3003"
+	}
+
+	rtConfig.ClientOneURL = addPrefixIfNeeded(rtConfig.ClientOneURL)
+	rtConfig.ClientTwoURL = addPrefixIfNeeded(rtConfig.ClientTwoURL)
+
+	return &rtConfig, nil
+}
+
+// getPaymailDomain retrieves the shared configuration from the SPV Wallet.
+func getPaymailDomain(xpub string, clientUrl string) (string, error) {
+	req, err := http.NewRequest(http.MethodGet, clientUrl+domainSuffixSharedConfig, nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Set(models.AuthHeader, xpub)
+	client := http.Client{
+		Timeout: timeoutDuration,
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to get shared config: %s", resp.Status)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	var configResponse models.SharedConfig
+	if err := json.Unmarshal(body, &configResponse); err != nil {
+		return "", err
+	}
+
+	if len(configResponse.PaymailDomains) != 1 {
+		return "", fmt.Errorf("expected 1 paymail domain, got %d", len(configResponse.PaymailDomains))
+	}
+
+	return configResponse.PaymailDomains[0], nil
```

Comment:
I just saw that, 
Why we don't use `client.GetSharedConfig`?

Reply:
changed to using `client.GetSharedConfig`